independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/multipart/QuarkusMultipartFormUpload.java

```diff
@@ -0,0 +1,202 @@
+package org.jboss.resteasy.reactive.client.impl.multipart;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.HttpConstants;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http.multipart.DefaultHttpDataFactory;
+import io.netty.handler.codec.http.multipart.FileUpload;
+import io.netty.handler.codec.http.multipart.HttpPostRequestEncoder;
+import io.netty.handler.codec.http.multipart.MemoryFileUpload;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.impl.headers.HeadersAdaptor;
+import io.vertx.core.streams.ReadStream;
+import io.vertx.core.streams.impl.InboundBuffer;
+import java.io.File;
+import java.nio.charset.Charset;
+
+/**
+ * based on {@link io.vertx.ext.web.client.impl.MultipartFormUpload}
+ */
+public class QuarkusMultipartFormUpload implements ReadStream<Buffer> {
```

Comment:
I sure hope this is a straight copy of `io.vertx.ext.web.client.impl.MultipartFormUpload`, otherwise it will be impossible to vouch for :)

Reply:
the only different thing is https://github.com/quarkusio/quarkus/pull/20127/files/106b744f930cdafb9f6e0cb545501bff9a2d9669#diff-4ceae636c50f58f65afb1a870e6887596574811b729893add5b4b13a64840fb9R73-R78

Reply:
Great!